### PR TITLE
Log exceptions thrown by HttpHandlers in repository integration tests

### PR DESCRIPTION
**Pull request by tlrx**
This pull request changes the `ESMockAPIBasedRepositoryIntegTestCase` so that `HttpHandler` are now wrapped in order to log any exceptions that could be thrown when executing the server side logic in repository integration tests.

This should (hopefully) help debugging recent test failures like:
https://gradle-enterprise.elastic.co/s/jkkkmi6mvuixk/tests/lgycqkw2bfw46-kysdm6ag7igo6
https://gradle-enterprise.elastic.co/s/3fv4hpmy3mulm/tests/lgycqkw2bfw46-clvf2uhrzpiw4
https://gradle-enterprise.elastic.co/s/rqyhtjrdui3oy/tests/lgycqkw2bfw46-clvf2uhrzpiw4

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESMockAPIBasedRepositoryIntegTestCase.java [hunk 1]
```diff
@@ -188,4 +183,21 @@ protected boolean canFailRequest(final HttpExchange exchange) {
...
+
+    /**
+     * Wrap a {@link HttpHandler} to log any thrown exception using the given {@link Logger}.
+     */
+    private static HttpHandler wrap(final HttpHandler handler, final Logger logger) {
+        return exchange -> {
+            try {
+                handler.handle(exchange);
+            } catch (final Exception e) {
+                String request = exchange.getRemoteAddress().toString()
```
Comment (original-brownbear) [inline test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESMockAPIBasedRepositoryIntegTestCase.java]: This is kind of weird? :) Why not just do

```java
logger.error(new ParametrizedMessage("Exception when handling request {} {} {}", exchange.getRemoteAddress(),  exchange.getRequestMethod(), exchange.getRequestURI()), e);
```

? :)

**Review by original-brownbear — APPROVED** [review]
LGTM, though maybe do the logging nicer? :)

**Comment by tlrx** [comment]
@elasticmachine run elasticsearch-ci/1
@elasticmachine run elasticsearch-ci/2

**Comment by tlrx** [comment]
@elasticmachine update branch

**Comment by tlrx** [comment]
Thanks @original-brownbear !

**Pull request merged**